### PR TITLE
Make checks for missing answers more strict

### PR DESCRIPTION
**Pull request by fernflower**
Previously only presence of one section in the answerfile
has been verified, which led to false-positive preupgrade
reports when some other key has been present in the
dialog section instead of the required one.

OAMG-7521

### CODE REVIEW

**Comment by fernflower** [comment]
@pirat89 This should be it.

File: leapp/messaging/__init__.py [hunk 1]
```diff
@@ -164,6 +169,7 @@ def register_dialog(self, dialog, actor):
                                      key=stable_key), actor)
         else:
```
[the remark this answers is not in the document]
Comment (pirat89) [reply leapp/messaging/__init__.py:172]: Discussed, it will be resolved properly in a follow up PR

**Review by pirat89 — APPROVED** [review]
lgtm and works as exepcted. Additional discovered issues (unrelated to this bug) will be handled properly as part of OAMG-7521 and OAMG-7525. Waiting for the finished CI test, preapred for merging

**Pull request merged**